app/controllers/app.py

```diff
@@ -1065,3 +1067,62 @@ def delete(self, app_id):
 
         except Exception as e:
             return dict(status='fail', message=str(e)), 500
+
+
+class AppMemoryUsageView(Resource):
+
+    @jwt_required
+    def post(self, app_id):
+        """
+        """
+
+        app_memory_schema = MetricsSchema()
+        app_query_data = request.get_json()
+
+        validated_query_data, errors = app_memory_schema.load(
+            app_query_data)
+
+        if errors:
+            return dict(status='fail', message=errors), 400
+
+        current_time = datetime.datetime.now()
+        yesterday_time = current_time + datetime.timedelta(days=-1)
+
+        start = validated_query_data.get('start', yesterday_time.timestamp())
+        end = validated_query_data.get('end', current_time.timestamp())
+        step = validated_query_data.get('step', '1h')
+
+        current_user_id = get_jwt_identity()
+        current_user_roles = get_jwt_claims()['roles']
+
+        app = App.get_by_id(app_id)
+
+        if not app:
+            return dict(status='fail', message=f'App {app_id} not found'), 404
+
+        project = app.project
+
+        if not project:
+            return dict(status='fail', message='Internal server error'), 500
+
+        if not is_owner_or_admin(project, current_user_id, current_user_roles):
+            return dict(status='fail', message='Unauthorised'), 403
+
+        app_alias = app.alias
+
+        prom_memory_data = prometheus.query_rang(
+            start=start,
+            end=end,
+            step=step,
+            metric='sum(rate(container_memory_usage_bytes{container_name!="POD",image!="",pod=~"'+app_alias+'.*"}[5m]))')
```

Comment:
You need to add a `namespace` to this metric

Reply:
okay then

Reply:
resolved